torch/_inductor/fx_passes/post_grad.py

```diff
@@ -80,6 +88,7 @@ def post_grad_passes(gm: torch.fx.GraphModule, is_inference: bool):
             patterns.apply(gm.graph)
         if is_inference:
             inference_patterns.apply(gm.graph)
+        triton_patterns.apply(gm.graph)
```

Comment:
Clones are removed as part of `remove_noop_ops`. Looking at the output code, I did not see the clones but perhaps we are changed.

Reply:
I think it's better to do this as part of `reinplace_scatters`.

1. The clone removal pass doesn't handle inplace mutations.
2. `reinplace_scatters` is exactly for this purpose - it converts `scatter` (i.e. clone + scatter_) into just `scatter_`. It also converts `scatter + copy_` into just `scatter_` as well.